### ISSUE TITLE
"disk I/O error"

### ISSUE THREAD

**Issue by blondie63** [tekartik/sqflite#335]
Hi, i've found same error of https://github.com/tekartik/sqflite/issues/71

Unhandled Exception: DatabaseException(Error Domain=FMDatabase Code=10 "disk I/O error" UserInfo={NSLocalizedDescription=disk I/O error}

To fix it simply stop app and run again..
But it's strange..

My Flutter Doctor:
[✓] Flutter (Channel stable, v1.12.13+hotfix.5, on Mac OS X 10.15.2 19C57, locale it-IT)
    • Flutter version 1.12.13+hotfix.5 at /Users/mauro/Documents/SDK/flutter
    • Framework revision 27321ebbad (2 weeks ago), 2019-12-10 18:15:01 -0800
    • Engine revision 2994f7e1e6
    • Dart version 2.7.0

 
[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.0-rc2)
    • Android SDK at /Users/mauro/Library/Android/sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-29, build-tools 29.0.0-rc2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3, Build version 11C29
    • CocoaPods version 1.8.4

[✓] Android Studio (version 3.5)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 42.1.1
    • Dart plugin version 191.8593
    • Java version OpenJDK Runtime Environment (build 1.8.0_202-release-1483-b49-5587405)

[✓] VS Code (version 1.41.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.7.1

[✓] Connected device (1 available)
    • iPad Pro (11-inch) • 271E614C-7A42-47BB-A95E-795BBDB20840 • ios • com.apple.CoreSimulator.SimRuntime.iOS-13-3 (simulator)

• No issues found!

i've archive 2.0.11

**Comment by alextekartik** [comment]
Thanks for the report. As noted in the initial bug, any information on how to reproduce it, whether you perform extra steps and how you open the database would be helpful. Thanks!

**Comment by davihoffmann** [comment]
same error. 

Are you solved it?

**Comment by blondie63** [comment]
Yes i've solved it but i don't remember how.. i think rebooting mac..

**Comment by kmvignesh** [comment]
I also faced this error. I resolved the issue by closing the database. In our case we where replacing the db file in certain condition. when that happens singleton database object was giving this error. So we closed the db connection when we do database replacement via code.

NOTE : we where getting this issue only on iOS. In Android without closing db its working fine.

Hope this helps.
